File: EcoNexo/scripts/delete_mock_events.ts
import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';
import path from 'path';

// Load env vars
dotenv.config({ path: path.resolve(process.cwd(), '.env.local') });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || '';
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || '';

const supabase = createClient(supabaseUrl, supabaseKey);

async function deleteMocks() {
  console.log("Deleting mock events from Supabase...");
  // notes contains 'Automated 2026 seeding'
  const { data, error } = await supabase
    .from('events')
    .delete()
    .like('notes', 'Automated 2026 seeding%');
    
  if (error) {
    console.error("Error deleting:", error);
  } else {
    console.log("Successfully deleted mock events.");
  }
}

deleteMocks().catch(console.error);

File: EcoNexo/scripts/delete_mock_events.py
import os
import requests

url = ""
key = ""

with open(".env.local", "r") as f:
    for line in f:
        line = line.strip()
        if line.startswith("NEXT_PUBLIC_SUPABASE_URL="):
            url = line.split("=", 1)[1].strip().strip('\'"')
        if line.startswith("NEXT_PUBLIC_SUPABASE_ANON_KEY="):
            key = line.split("=", 1)[1].strip().strip('\'"')

print(f"Loaded url: {bool(url)}, key: {bool(key)}")

if url and key:
    headers = {
        "apikey": key,
        "Authorization": f"Bearer {key}"
    }
    
    # Supabase REST API eq operator
    res = requests.delete(
        f"{url}/rest/v1/events?notes=like.Automated 2026 seeding*",
        headers=headers
    )
    print("Delete status:", res.status_code)
    print("Delete text:", res.text)
else:
    print("Still missing credentials")
